Update metricflow-time-spine.md (#6097)
dbt-labs/docs.getdbt.com: website/docs/docs/build/metricflow-time-spine.md

MetricFlow requires you to define a time spine table as a project-level configuration for time-based joins and aggregations, like cumulative metrics. You only need to configure time spine models that the Semantic Layer should recognize. At a minimum, define a time spine table for a daily grain. You can optionally define a time spine table for a different granularity, like hourly. Note that if you don’t have a date or calendar model in your project, you'll need to create one. 

If you already have a date dimension or time spine table in your dbt project, you can point MetricFlow to this table by updating the `model` configuration to use this table in the Semantic Layer. This is a model-level configuration that tells dbt to use the model for time range joins in the Semantic Layer. 

> For example, given the following directory structure, you can create two time spine configurations, `time_spine_hourly` and `time_spine_daily`.  MetricFlow supports granularities ranging from milliseconds to years, refer to the [Dimensions page](/docs/build/dimensions#time?dimension=time_granularity) to find the full list of supported granularities.

Author: ```suggestion
For example, given the following directory structure, you can create two time spine configurations, `time_spine_hourly` and `time_spine_daily`.  MetricFlow supports granularities ranging from milliseconds to years, refer to the [Dimensions page](/docs/build/dimensions?dimension=time_gran#time) (time_granularity tab) to find the full list of supported granularities.
```

MetricFlow requires you to define a time spine table as a project-level configuration for time-based joins and aggregations, like cumulative metrics. 
- You only need to configure time spine models that the Semantic Layer should recognize.
- At a minimum, define a time spine table for a daily grain.
- You can optionally define a time spine table for a different granularity, like hourly.
- Note that if you don’t have a date or calendar model in your project, you'll need to create one. 

If you already have a date dimension or time spine table in your dbt project, you can point MetricFlow to this table by updating the `model` configuration to use this table in the Semantic Layer. This is a model-level configuration that tells dbt to use the model for time range joins in the Semantic Layer. 

For example, given the following directory structure, you can create two time spine configurations, `time_spine_hourly` and `time_spine_daily`.  MetricFlow supports granularities ranging from milliseconds to years, refer to the [Dimensions page](/docs/build/dimensions?dimension=time_gran#time) (time_granularity tab) to find the full list of supported granularities.

:::tip 

> Previously, you were required to create a model called `metricflow_time_spine` in your dbt project. This is no longer required. However, you can build your time spine model from this table if you don't have another date dimension table you want to use in your project.

Author: ```suggestion
Previously, you had to create a model called `metricflow_time_spine` in your dbt project. Now, if your project already includes a date dimension or time spine table, you can simply configure MetricFlow to use that table by updating the `model` setting in the Semantic Layer.

If you don’t have a date dimension table, you can still create one by using the code snippet below to build your time spine model.
```

tags: [Metrics, Semantic Layer]
---

It's common in analytics engineering to have a date dimension or "time spine" table as a base table for different types of time-based joins and aggregations. The structure of this table is typically a base column of daily or hourly dates, with additional columns for other time grains, like fiscal quarters, defined based on the base column. You can join other tables to the time spine on the base column to calculate metrics like revenue at a point in time, or to aggregate to a specific time grain.

> MetricFlow requires you to define a time spine table as a project-level configuration for time-based joins and aggregations, like cumulative metrics. 

Author: @Jstein77 confirming timespine is for model-level  config adn not project-level
```suggestion
MetricFlow requires you to define a time spine table as a model-level configuration in the Semantic Layer for time-based joins and aggregations, such as cumulative metrics. This configuration informs dbt which model should be used for time range joins. It is especially useful for cumulative metrics or calculating time-based offsets.
```

Reviewer: Yea this is more clear. I would update this to sentence to:

> It is especially useful for cumulative metrics or calculating time-based offsets.
The time-spine model is joined to other tables when calculating certain types of metrics or dimensions. MetricFlow will join to the time-spine model in the compiled SQL for the following types of metrics and dimensions:
* Cumulative metrics
* Metric offsets
* Conversion metrics
* Slowly Changing Dimensions
* Metrics with the join_to_timespine configuration set to true
To see the generated SQL for the metric and dimension types that utilize time spine joins refer to the docs, or add the `compile=True` flag when querying the semantic layer to return the compiled SQL.

- You only need to configure time spine models that the Semantic Layer should recognize.
- At a minimum, define a time spine table for a daily grain.
- You can optionally define a time spine table for a different granularity, like hourly.

> - Note that if you don’t have a date or calendar model in your project, you'll need to create one. 

Author: ```suggestion
- Note that if you don’t have a date or calendar model in your project, you'll need to create one. 
- If you're looking to specify the grain of a time dimension so that MetricFlow can transform the underlying column to the required granularity, refer to the [Time granularity documentation](/docs/build/dimensions?dimension=time_gran)
```

MetricFlow requires you to define a time spine table as a project-level configuration for time-based joins and aggregations, like cumulative metrics. 
- You only need to configure time spine models that the Semantic Layer should recognize.
- At a minimum, define a time spine table for a daily grain.
- You can optionally define a time spine table for a different granularity, like hourly.
- Note that if you don’t have a date or calendar model in your project, you'll need to create one. 
- If you're looking to specify the grain of a time dimension so that MetricFlow can transform the underlying column to the required granularity, refer to the [Time granularity documentation](/docs/build/dimensions?dimension=time_gran)

If you already have a date dimension or time spine table in your dbt project, you can point MetricFlow to this table by updating the `model` configuration to use this table in the Semantic Layer. This is a model-level configuration that tells dbt to use the model for time range joins in the Semantic Layer. 

> For example, given the following directory structure, you can create two time spine configurations, `time_spine_hourly` and `time_spine_daily`.  MetricFlow supports granularities ranging from milliseconds to years, refer to the [Dimensions page](/docs/build/dimensions?dimension=time_gran#time) (time_granularity tab) to find the full list of supported granularities.

Author: ```suggestion
For example, given the following directory structure, you can create two time spine configurations, `time_spine_hourly` and `time_spine_daily`.  MetricFlow supports granularities ranging from milliseconds to years. Refer to the [Dimensions page](/docs/build/dimensions?dimension=time_gran#time) (time_granularity tab) to find the full list of supported granularities.
```
